main.c

```diff
@@ -48,20 +47,61 @@ char type_to_char(enum tile_type type);
 int main(void) {
 
     printf("Welcome to Frogger Game!\n");
+
     struct board_tile game_board[SIZE][SIZE];
 
-    // TODO (Stage 1.1) Initialise the gameboard.
-    
-    // Read user input and place turtles.
+    // Initialize the game board
+    initialize_board(game_board);
+
+    // Read user input and place turtles
+    int num_turtles;
     printf("How many turtles? ");
-    // TODO (Stage 1.2): Scan in the turtles, and place them on the map.
+    scanf("%d", &num_turtles);
+    place_turtles(game_board, num_turtles);
 
-    // Start the game and print out the gameboard.
+    // Start the game and print out the gameboard
     printf("Game Started\n");
     print_board(game_board);
 
-    printf("Enter command: ");
-    // TODO (Stage 1.3): Create a command loop, to read and execute commands!
+    // Enter command loop
+    char command;
+    int row, start_col, end_col;
+
+    while (1) {
+        printf("Enter command: ");
+        int result = scanf(" %c", &command);
+        
+        // Exit if CTRL+C or input is invalid
+        if (result == EOF || command == 'q') break;
+        
+        // Handle log command
+        if (command == 'l') {
+            scanf("%d %d %d", &row, &start_col, &end_col);
+
+            // Ensure the row and columns are within bounds and check if row has no turtles
+            if (row > 0 && row < SIZE - 1 && start_col < SIZE && end_col >= 0) {
+                int has_turtle = FALSE;
+                for (int col = 0; col < SIZE; col++) {
```

Comment:
```suggestion
                for (int col = startCol; col <= endCol; col++) {
```

for checking the turtle it should check by the row and each column between start and end, so you initialize the loop starting from the `startCol` and ending at the `endCol`